### PR TITLE
feat: add content block on generate_hcl 

### PR DESCRIPTION
**Pull request by katcipis**
This PRs changes the behavior of code generation using `generate_hcl`. Instead of generating code by doing:

```hcl
generate_hcl {
  terraform {
  }
}
```

All generated HCL must be inside the `content` block, like this:

```hcl
generate_hcl {
  content {
    terraform {
    }
  }
}
```

A full example:

```sh
#!/bin/bash

set -o nounset
set -o errexit

basedir=$(mktemp -d)

cd "${basedir}"

# We need the project root config since it is not git
cat > terramate.tm.hcl <<- EOM
terramate {
  config {
  }
}
EOM

mkdir -p stacks/stack-{1,2}
terramate stacks init stacks/stack-{1,2}

# Defining some code for all stacks
# Globals are defined on root and also on stacks, to showcase possibilities
# Inspired on: https://github.com/mineiros-io/terramate-blueprint-iac-github/tree/main/stacks/repositories/teamA
cat > stacks/terramate.tm.hcl <<- EOM
globals {
    tf_version = "1.1.3"
    gh_owner = "the-awesome-mineiros"
    gh_provider_version = "4.19.1"
}

generate_hcl "provider.tf" {
    content {
        provider "github" {
          owner = global.gh_owner
        }

        terraform {
          required_providers {
            github = {
              source  = "integrations/github"
              version = global.gh_provider_version
            }
          }
        }

        terraform {
          required_version = global.tf_version
        }
    }
}

generate_hcl "main.tf" {
  content {
      module "repositories" {
        source = "../../../modules/terramate-github-repositories"

        repositories        = global.repositories
        repository_defaults = global.repository_defaults
        member_map          = global.member_map
      }
  }
}
EOM

# Now add some per stack globals

cat > stacks/stack-1/terramate.tm.hcl <<- EOM
globals {
    repositories = [
        {
            name = "stack-1"
            visibility = "private"
        }
    ]
    repository_defaults = {
        config = "stack-1"
    }
    member_map  = {
        "first.lastname" = "powerpuffGirls69"
    }
}
stack {}
EOM

cat > stacks/stack-2/terramate.tm.hcl <<- EOM
globals {
    repositories = [
        {
            name = "stack-2"
            visibility = "public"
        }
    ]
    repository_defaults = {
        config = "stack-2"
    }
    member_map  = {
        "first.lastname" = "onePunchMan666"
    }
}
stack {}
EOM


echo "initial tree"
tree

echo
echo "generating code for all stacks"
terramate generate

echo "tree after generating code"
tree

echo
echo "checking generated main.tf files"
terramate run -- cat main.tf

echo
echo "checking generated provider.tf files"
terramate run -- cat provider.tf
```

### CODE REVIEW

**Review by mariux — APPROVED** [review]
lgtm

**Comment by mariux** [comment]
tested works like a charm...please ping once merged so i can merge https://github.com/mineiros-io/terramate-blueprint-iac-github/pull/18

File: generate/genhcl/genhcl.go [hunk 1]
```diff
@@ -239,6 +254,16 @@ func validateGenerateHCLBlock(block *hclsyntax.Block) error {
...
+	if len(block.Body.Blocks) != 1 {
+		return fmt.Errorf("one 'content' block is required, got %d blocks", len(block.Body.Blocks))
+	}
+	contentBlock := block.Body.Blocks[0]
+	if contentBlock.Type != "content" {
+		return fmt.Errorf("one 'content' block is required, got %q block", contentBlock.Type)
+	}
+	if len(contentBlock.Labels) > 0 {
+		return fmt.Errorf("content block has unexpected labels: %v", contentBlock.Labels)
+	}
```
Comment (mariux) [inline generate/genhcl/genhcl.go:266]: why do we manually check those... can't we use the go structures to let HCL library check this for us?

File: docs/hcl-generation.md [hunk 2]
```diff
@@ -62,22 +67,27 @@ in the root configuration:
 ```hcl
 generate_hcl "provider.tf" {
 
-  provider "name" {
-    param = global.provider_data
-  }
+  content {
+
```
Comment (i4ki) [inline docs/hcl-generation.md:71]:
```suggestion
```

**Review by i4ki — APPROVED** [review]
lgtm

**Review by i4ki — APPROVED** [review]

File: generate/genhcl/genhcl.go (hunk 1 above)
Comment (katcipis) [reply generate/genhcl/genhcl.go:266]: When you mentioned "Go Structures" I immediately thought of [hclsimple](https://pkg.go.dev/github.com/hashicorp/hcl/v2/hclsimple).

Which in this case seems less helpful IMO (but quite helpful on some scenarios), but now I think maybe you meant the [BodySchema](https://pkg.go.dev/github.com/hashicorp/hcl2@v0.0.0-20191002203319-fb75b3253c80/hcl#BodySchema).

Which maybe could make some of the basic validation for us, will take a look (haven't used it yet, sorry for the ignorance).

**Pull request merged**